packages/compass-data-modeling/src/utils/nodes-and-edges.spec.tsx

```diff
@@ -0,0 +1,475 @@
+import React from 'react';
+import { expect } from 'chai';
+import {
+  screen,
+  waitFor,
+  render,
+  userEvent,
+} from '@mongodb-js/testing-library-compass';
+import { getFieldsFromSchema } from '../utils/nodes-and-edges';
+
+describe('getFieldsFromSchema', function () {
+  const validateMixedType = async (
```

Comment:
Not a big deal, but I almost feel like having the tooltip as part of the output `getFieldsFromSchema` is mixing too much UI logic into the method, maybe it's better to split the two (but it's not a big deal either, just something to keep in mind for interfaces like that in the future maybe)